pkg/controllers/disruption/orchestration/queue.go

```diff
@@ -305,53 +309,57 @@ func (q *Queue) waitOrTerminate(ctx context.Context, cmd *Command) error {
 // Add adds commands to the Queue
 // Each command added to the queue should already be validated and ready for execution.
 func (q *Queue) Add(cmd *Command) error {
-	providerIDs := lo.Map(cmd.candidates, func(s *state.StateNode, _ int) string {
-		return s.ProviderID()
-	})
 	// First check if we can add the command.
-	if q.HasAny(providerIDs...) {
+	if q.HasAny(cmd.candidates...) {
 		return fmt.Errorf("candidate is being disrupted")
 	}
 
+	q.Lock()
+	defer q.Unlock()
 	cmd.timeAdded = q.clock.Now()
-	q.mu.Lock()
-	for _, candidate := range cmd.candidates {
-		q.providerIDToCommand[candidate.ProviderID()] = cmd
+	for _, c := range cmd.candidates {
+		q.nodeClaimsInQueueSet.Insert(c.NodeClaim.Status.ProviderID)
```

Comment:
I don't have a strong opinion, but it seems like you _could_ just keep the same paradigm as before -- I don't quite see the reason to change it up since the command is a pointer anyways